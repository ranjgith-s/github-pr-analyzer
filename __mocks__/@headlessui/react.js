// __mocks__/@headlessui/react.js
const React = require('react');

function passthrough({ children, ...props }) {
  return React.createElement('div', props, children);
}

module.exports = {
  __esModule: true,
  Card: passthrough,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  HeroUIProvider: passthrough,
};
