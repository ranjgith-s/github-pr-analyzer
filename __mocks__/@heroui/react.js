const React = require('react');

// List of custom props to filter out for DOM elements
const CUSTOM_PROPS = [
  'isCurrent',
  'endContent',
  'onPress',
  'variant',
  'color',
  'size',
  'underline',
  'src',
  'alt',
  'className',
  'children',
];

function filterProps(props) {
  const domProps = { ...props };
  CUSTOM_PROPS.forEach((key) => {
    if (key in domProps && key !== 'className' && key !== 'children') {
      delete domProps[key];
    }
  });
  return domProps;
}

function passthrough({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: props.className }, children);
}

function InputMock(props) {
  return React.createElement('input', filterProps(props));
}

function ButtonMock({ children, onPress, onClick, ...props }) {
  // Map onPress to onClick for DOM, do not pass onPress to DOM
  const handler = onPress || onClick;
  const domProps = { ...filterProps(props), onClick: handler, className: props.className };
  return React.createElement('button', domProps, children);
}

function Spinner() {
  // Only return a simple span for test, not a div with data-testid
  return React.createElement('span', null, 'Loading...');
}

function CardMock({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: (props.className || '') + ' mock-card' }, children);
}

module.exports = {
  __esModule: true,
  Card: CardMock,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  Button: ButtonMock,
  Input: InputMock,
  Avatar: passthrough,
  Breadcrumbs: passthrough,
  BreadcrumbItem: passthrough,
  Spinner,
};
