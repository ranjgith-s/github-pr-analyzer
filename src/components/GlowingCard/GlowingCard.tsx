import React from 'react';
import { Card } from '../ui';

interface Props {
  children: React.ReactNode;
  className?: string;
  [key: string]: any; // allow data-testid and other props
}

export default function GlowingCard({ children, className, ...rest }: Props) {
  return (
    <Card
      className={`bg-gradient-to-br from-blue-50 to-indigo-100 rounded-2xl shadow-xl p-8 min-w-[320px] max-w-[400px] mx-auto ${className || ''}`}
      {...rest}
    >
      {children}
    </Card>
  );
}
