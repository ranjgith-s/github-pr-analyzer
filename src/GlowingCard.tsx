import React from 'react';
import { Box } from '@primer/react';

interface Props {
  children: React.ReactNode;
  className?: string;
}

export default function GlowingCard({ children, className }: Props) {
  return (
    <Box
      className={`glow-card${className ? ` ${className}` : ''}`}
      sx={{
        p: 4,
        width: '100%',
        maxWidth: 400,
        position: 'relative',
        borderRadius: 2,
        overflow: 'visible',
      }}
    >
      {children}
    </Box>
  );
}
