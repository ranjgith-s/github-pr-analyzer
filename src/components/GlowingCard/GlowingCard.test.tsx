import React from 'react';
import { render, screen } from '@testing-library/react';
import GlowingCard from '../GlowingCard';

describe('GlowingCard', () => {
  it('renders children correctly', () => {
    render(
      <GlowingCard>Test Content</GlowingCard>
    );
    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });

  it('applies custom className', () => {
    render(
      <GlowingCard className="custom-class" data-testid="glow-card">Content</GlowingCard>
    );
    const card = screen.getByTestId('glow-card');
    expect(card).toHaveClass('custom-class');
  });

  it('matches snapshot', () => {
    const { container } = render(<GlowingCard>Snapshot</GlowingCard>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
