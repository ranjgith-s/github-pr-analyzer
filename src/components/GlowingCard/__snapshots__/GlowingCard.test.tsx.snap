// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlowingCard matches snapshot 1`] = `
<div
  class="border bg-card text-card-foreground bg-gradient-to-br from-blue-50 to-indigo-100 rounded-2xl shadow-xl p-8 min-w-[320px] max-w-[400px] mx-auto"
>
  Snapshot
</div>
`;
